d2l-search-facets-option.html

```diff
@@ -0,0 +1,97 @@
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../d2l-inputs/d2l-input-checkbox.html">
+
+<dom-module id="d2l-search-facets-option">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: inline-block;
+			}
+			.d2l-search-facets-option-checkbox {
+				@apply --d2l-body-compact-text;
+			}
+		</style>
+		<d2l-input-checkbox
+			checked={{checked::change}}
+			disabled=[[disabled]]
+			name="[[name]]"
+			on-change="_handleChange"
+			class="d2l-search-facets-option-checkbox"
+		>[[_facetText]]</d2l-input-checkbox>
+	</template>
+	<script>
+		/**
+		 * `<d2l-search-facets-option>`
+		 * Search facet option, with a count to display the number of results
+		 */
+		class SearchFacetsOption extends Polymer.Element {
+			static get is() { return 'd2l-search-facets-option'; }
+			static get properties() {
+				return {
+					/**
+					* The text displayed for this option. Adds the count next to the text if present
+					*/
+					_facetText: {
+						type: String,
+						value: '',
+						computed: '_getFacetText(text, count)'
+					},
+					/**
+					* Whether the option is selected or not. This property matches the value of
+					* the d2l-input-checkbox
+					*/
+					checked: {
+						type: Boolean,
+						reflectToAttribute: true,
+					},
+					/**
+					* The number of search results for this option
+					*/
+					count: {
+						type: String,
```

Comment:
That could be an interesting feature that the component does on its own... like you could set a threshold after which it would say `10+` or `100+`? Not sure about this, using a string and losing the strongly-typed data is unfortunate.

Reply:
Generally for facetted search, the UI just shows exact numbers:
![image](https://user-images.githubusercontent.com/8430763/47461871-d81fa000-d7af-11e8-9595-1cd9b77c01a8.png)
[Library Search Example](https://primo.tug-libraries.on.ca/primo_library/libweb/action/search.do?fn=search&ct=search&initialSearch=true&mode=Basic&tab=search&indx=1&dum=true&srt=rank&vid=WATERLOO&frbg=&tb=t&vl%28freeText0%29=electricity&scp.scps=scope%3A%28%22CENTRAL%22%29%2Cscope%3A%28GUELPH%29%2Cscope%3A%28LAURIER%29%2Cscope%3A%28WATERLOO%29%2Cprimo_central_multiple_fe)

I can see possible use cases where DEV constraints might need to return an approximate number, but not sure if there's cases where Design would specify 10+ or the like. I would lean towards using number and loosening the constraint in the future IF we need it.

From other design constraint standpoint, if the number is 5,012,123,123  ( (5B+) (??) ) -- we might not want to show the entire number since it could get sorta ugly with weird wrapping. But NOT showing the number could also be confusing in a selection scenario. Again, can't imagine what data would result in 5 Billion, so I would lean towards number only.